### ISSUE TITLE
[Edge] TypeError: Unable to get property 'define' of undefined or null reference

### ISSUE THREAD

**Issue by TangoPJ** [sneas/img-comparison-slider#54]
Good day. I'm using this slider on my website and when I visit my website via Edge browser I get the following error:
```javascript
TypeError: Unable to get property 'define' of undefined or null reference
```
Could you use a polyfill to avoid common errors?
For example:
```javascript
defineElement = function( tag, opt ){
    if( typeof customElements !== 'undefined' ){
        customElements.define( tag, opt );
    } else {
        document.registerElement( tag, opt );
    }
}
class HTMLxSearchElement extends HTMLElement{
    constructor(){
        super();
        console.log('super');
    }
}
defineElement( 'x-search', HTMLxSearchElement )
```

Thank you a lot.

**Comment by sneas** [comment]
Hi @TangoPJ , thank you for the issue. I've just tried to open the library in the latest Edge browser and it seems working fine (both Mac and Windows). What version are you using?

**Comment by TangoPJ** [comment]
Yep you are right, version of the browser is 17.. It's too old I know, but I get this error from sentry, and used browserstack to reproduce the error

**Comment by sneas** [comment]
Uhhh, I need to think of a solution a bit.

I don't think the suggested polyfill will solve the problem of the old Edge not supporting web components. Unfortunately, I don't even have a way to test/debug that.

The only things I can suggest for now would be:
* use polyfills on your side https://www.webcomponents.org/polyfills
* never mind ancient browsers. There is always someone with Internet Explorer 6

Since you have access to the old Edge, could you try to run this code before importing img-comparison-slider?

```js
if( !window.customElements ){
  window.customElements = {
    define: document.registerElement.bind(document)
  };
}
```

The above code is the equivalent of the requested solution, but without changing the existing library. I doubt it'll help, but it's worth trying anyway.

**Comment by sneas** [comment]
Btw, I remember the feature you've requested. https://github.com/sneas/img-comparison-slider/issues/53. I'm going to release it this weekend.

**Comment by TangoPJ** [comment]
[#53](https://github.com/sneas/img-comparison-slider/issues/53) It will great feature! :)
Thank you for the quick answer, I will try to do it as you advised

**Comment by alexdanylyschyn** [comment]
I just took a look at this. The custom elements polyfill won't be enough to make this library work. The old Edge doesn't support Shadow DOM either, so 2 polyfills would be needed.
I don't think polyfills should be part of this library. So maybe we could add a requirements section in the README, where we reference https://www.webcomponents.org/polyfills
@sneas I could setup a small PR to add this if you agree?

**Comment by sneas** [comment]
That will be great, thanks, mate.

You're absolutely right, polyfills shouldn't be part of this library.
